sql/rowexec/dml_iters.go

```diff
@@ -514,6 +516,80 @@ type accumulatorIter struct {
 	updateRowHandler accumulatorRowHandler
 }
 
+func getRowHandler(clientFoundRowsToggled bool, iter sql.RowIter) accumulatorRowHandler {
+	switch i := iter.(type) {
+	case *plan.TableEditorIter:
+		return getRowHandler(clientFoundRowsToggled, i.InnerIter())
+	case *plan.CheckpointingTableEditorIter:
+		return getRowHandler(clientFoundRowsToggled, i.InnerIter())
+	case *ProjectIter:
+		return getRowHandler(clientFoundRowsToggled, i.childIter)
+	case *triggerIter:
+		return getRowHandler(clientFoundRowsToggled, i.child)
+	case *blockIter:
+		return getRowHandler(clientFoundRowsToggled, i.repIter)
+	case *insertIter:
+		if i.replacer != nil {
+			return &replaceRowHandler{}
+		}
+		if i.updater != nil {
+			return &onDuplicateUpdateHandler{schema: i.schema, clientFoundRowsCapability: clientFoundRowsToggled}
+		}
+		return &insertRowHandler{}
+	case *deleteIter:
+		return &deleteRowHandler{}
+	case *updateIter:
+		// it's possible that there's an updateJoinIter that's not the immediate child of updateIter
+		rowHandler := getRowHandler(clientFoundRowsToggled, i.childIter)
+		if rowHandler != nil {
+			return rowHandler
+		}
+		sch := i.schema
+		// special case for foreign keys,   plan.ForeignKeyHandler.Schema() returns original schema
+		if fkHandler, isFk := i.updater.(*plan.ForeignKeyHandler); isFk {
+			sch = fkHandler.Sch
+		}
+		return &updateRowHandler{schema: sch, clientFoundRowsCapability: clientFoundRowsToggled}
+	case *updateJoinIter:
+		rowHandler := &updateJoinRowHandler{
+			joinSchema: i.joinSchema,
+			tableMap:   plan.RecreateTableSchemaFromJoinSchema(i.joinSchema),
+			updaterMap: i.updaters,
+		}
+		i.accumulator = rowHandler
+		return rowHandler
+	default:
+		return nil
+	}
+}
+
+func AddAccumulatorIter(ctx *sql.Context, iter sql.RowIter) (sql.RowIter, sql.Schema) {
+	switch i := iter.(type) {
+	case sql.CustomRowIter:
+		childIter := i.GetChildIter()
+		childIter, sch := AddAccumulatorIter(ctx, childIter)
+		return i.SetChildIter(childIter), sch
+	case *callIter:
+		childIter, sch := AddAccumulatorIter(ctx, i.innerIter)
+		i.innerIter = childIter
+		return i, sch
+	case *beginEndIter:
```

Comment:
should these two implement the new interface?

Reply:
That would make sense; will change